positron/modules/atom_browser_web_contents.js

```diff
@@ -29,6 +32,57 @@ let WebContents_prototype = {
   loadURL: function(url) {
     this._browserWindow._domWindow.location = url;
   },
+
+  openDevTools: function() {
+    // TODO: When tools can be opened inside the content window, support
+    // `detach` option to force into a new window instead.
+
+    // Ensure DevTools core modules are loaded, including support for the about
+    // URL below which is registered dynamically.
+    const { require } = Cu.import("resource://devtools/shared/Loader.jsm", {});
+    require("devtools/client/main");
+
+    // XXX: Consider switching to WindowHost.  The current approach below avoids
+    // the need for a container window wrapping a tools frame, but it does
+    // replicate close handling, etc.  For the moment, having a custom path here
+    // seems more flexible as we figure out exactly what we actually want.
+    let window = this._browserWindow._domWindow;
+    let id = window.QueryInterface(Ci.nsIInterfaceRequestor)
+                   .getInterface(Ci.nsIDOMWindowUtils)
+                   .outerWindowID;
+    let url = `about:devtools-toolbox?type=window&id=${id}`;
+    let features = "chrome,resizable,centerscreen," +
+                   "width=1024,height=768";
+    let toolsWindow = Services.ww.openWindow(null, url, null, features, null);
+
+    let onLoad = () => {
+      toolsWindow.removeEventListener("load", onLoad);
+      toolsWindow.addEventListener("unload", onUnload);
+      window.addEventListener("unload", onBrowserUnload);
+      this.emit("devtools-opened");
+    }
+    let onUnload = () => {
+      toolsWindow.removeEventListener("unload", onUnload);
+      toolsWindow.removeEventListener("message", onMessage);
+      window.removeEventListener("unload", onBrowserUnload);
+      this.emit("devtools-closed");
+    }
+    let onMessage = ({ data }) => {
+      data = JSON.parse(data);
+      if (data.name !== "toolbox-close") {
+        return;
+      }
+      toolsWindow.close();
+    };
+    // TODO: Improve browser window close handling.  DevTools is currently still
+    // attached when the window closes, leading to large errors.
```

Comment:
On OS X at least for me, when I quit Positron (Cmd-Q) with both app window and toolbox open, the DevTools actors are still bound to the app window, and they get a window destroyed notification, but then try to keep using the window which no longer exists, and explode in various ways.